### ISSUE TITLE
cask-repair ignores version

### ISSUE THREAD

**Issue by joe776** [vitorgalvao/tiny-scripts#135]
I tried to update the eclipse-jee cask to 4.8 Photon. From what I got form the existing formula, the correct way to specify the version should be 4.8.0,photon:R. However, cask-repair simply ignores this provided version and downloads the old 4.7.3a, no matter if entered in the prompt or via cask-repair -v "4.8.0,photon:R" eclipse-jee. The editor then opening also still shows the old version.

I ran `brew reinstall vitorgalvao/tiny-scripts/cask-repair` to make sure I have the latest version of the script (already was the latest).
I'm using the non-prefixed version of coreutils. Can this cause any issues?
I put a quick `echo` inside the `modify_stanza` function, and the `new_stanza_value` is picked up there. It looks like the edit through perl fails. Is there an easy way to debug this? Perl looks also up-to-date with version 5.28.

**Comment by vitorgalvao** [comment]
Like I said I can’t reproduce, so I have no way to look into the problem and fix it.

I will review a PR that does, though.

**Comment by joe776** [comment]
The perl command that is executed to change the version looks like this:
```
perl -0777 -i -e'
    $last_stanza_match = shift(@ARGV);
    $stanza_start = shift(@ARGV);
    $new_stanza_value = shift(@ARGV);
    $ending_comma = shift(@ARGV);
    print <> =~ s|\Q$last_stanza_match\E|$stanza_start$new_stanza_value$ending_comma|r;
' "version '4.8.0,photon:R'" "version" "'4.9.0'" "" "/usr/local/Homebrew/Library/Taps/homebrew/homebrew-cask/Casks/eclipse-jee.rb"
```
The formular ends up unchanged. I also tried it on another formula, or escaping the backslashes in the regexp with a second one, same result.
I don't know any perl, so should this command look that way? Running it manually also produces no output.

I used the cask-repair script back in December, when it worked without problems.

**Comment by joe776** [comment]
```
perl --version
This is perl 5, version 28, subversion 0 (v5.28.0) built for darwin-thread-multi-2level
```
```
bash --version
GNU bash, Version 4.4.23(1)-release (x86_64-apple-darwin17.5.0)
```

macOS is the latest High Sierra, 10.13.5

I ran the command `perl -0777 -i -e 'print <> =~ s/\Qversion '4.8.0,photon:R'\E/version 4.9.0/r;' /usr/local/Homebrew/Library/Taps/homebrew/homebrew-cask/Casks/eclipse-jee.rb` directly in the bash, but the version still stays as `version '4.8.0,photon:R'`.
Basically the same RE with sed works: `sed -i 's/version '4.8.0,photon:R'/version 4.9.0/g' /usr/local/Homebrew/Library/Taps/homebrew/homebrew-cask/Casks/eclipse-jee.rb`.

**Comment by vitorgalvao** [comment]
Fixed in #137 by @commitay. Already updated the formula, so `brew upgrade cask-repair` will fix it.

**Comment by joe776** [comment]
Thanks! Works here, too.
